bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.openhab.binding.publictransportswitzerland.internal.PublicTransportSwitzerlandBindingConstants.*;
+
+/**
+ * The {@link PublicTransportSwitzerlandStationboardHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeremy Stucki - Initial contribution
+ */
+@NonNullByDefault
+public class PublicTransportSwitzerlandStationboardHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PublicTransportSwitzerlandStationboardHandler.class);
+
+    private @Nullable ScheduledFuture<?> updateDataJob;
+
+    // Limit the API response to the necessary fields
+    private final String fieldFilters = createFilterForFields(
+            "stationboard/to",
+            "stationboard/category",
+            "stationboard/number",
+            "stationboard/stop/departureTimestamp",
+            "stationboard/stop/delay",
+            "stationboard/stop/platform");
+
+    public PublicTransportSwitzerlandStationboardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // TODO?
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        //noinspection ConstantConditions
+        if (updateDataJob == null || updateDataJob.isCancelled()) {
+            updateDataJob = scheduler.scheduleWithFixedDelay(this::updateData, 0, 60, TimeUnit.SECONDS);
+        }
+    }
+
+    public void updateData() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        try {
+            String escapedStation = URLEncoder.encode(config.station, StandardCharsets.UTF_8.name());
+            String requestUrl = BASE_URL + "stationboard?station=" + escapedStation + fieldFilters;
+
+            String response = HttpUtil.executeUrl("GET", requestUrl, 10_000);
+            JsonElement jsonObject = new JsonParser().parse(response);
+
+            updateCsvChannel(jsonObject);
+            updateJsonChannel(jsonObject);
```

Comment:
I don't believe an end user needs to know the JSON response from the API.
I think you could replace this channel by a DEBUG log that will log the JSON response.

Reply:
Resolved with e463aac3c9